### PR TITLE
Fixes some defects in this demo

### LINKED ISSUE #6783: Contoso University Demo Broken
When running through the Contoso University example application for Windows we are running into errors starting the app. We need to update our instructions to add some steps so this application will start up as expected.

- [x] The OS (including version) where you are running any of the Habitat commands.

Windows 10 Professional with Docker

To get this demo working again we need to make the following changes to our tutorial:

- [x] The `init` hook needs to be updated to show the following:
```powershell
Set-Location {{pkg.svc_path}}
if(Test-Path var) { Remove-Item var -Recurse -Force }
New-Item -Name var -ItemType Junction -target "{{pkg.path}}/www" | Out-Null
Set-Location {{pkg.svc_path}}\var
New-Item -Name Web.config -ItemType SymbolicLink -target "{{pkg.svc_config_path}}/Web.config" -Force | Out-Null
```
- [x] A `web.config` needs to be added to the config dir in step 5:
```xml
<?xml version="1.0" encoding="utf-8"?>
<!--
  For more information on how to configure your ASP.NET application, please visit
  http://go.microsoft.com/fwlink/?LinkId=301880
  -->
<configuration>
  <configSections>
    <!-- For more information on Entity Framework configuration, visit http://go.microsoft.com/fwlink/?LinkID=237468 -->
    <section name="entityFramework" type="System.Data.Entity.Internal.ConfigFile.EntityFrameworkSection, EntityFramework, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089" requirePermission="false" />
  </configSections>
  <connectionStrings>
    <add name="SchoolContext" connectionString="Data Source={{bind.database.first.sys.ip}},{{bind.database.first.cfg.port}};Initial Catalog=ContosoUniversity2;User ID={{bind.database.first.cfg.username}};Password={{bind.database.first.cfg.password}};" providerName="System.Data.SqlClient" />
  </connectionStrings>
  <appSettings>
    <add key="webpages:Version" value="3.0.0.0" />
    <add key="webpages:Enabled" value="false" />
    <add key="ClientValidationEnabled" value="true" />
    <add key="UnobtrusiveJavaScriptEnabled" value="true" />
  </appSettings>
  <system.web>
    <compilation debug="true" targetFramework="4.5" />
    <httpRuntime targetFramework="4.5" />
    <httpModules>
    </httpModules>
  </system.web>
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Newtonsoft.Json" culture="neutral" publicKeyToken="30ad4fe6b2a6aeed" />
        <bindingRedirect oldVersion="0.0.0.0-6.0.0.0" newVersion="6.0.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Web.Optimization" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="1.0.0.0-1.1.0.0" newVersion="1.1.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="WebGrease" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="0.0.0.0-1.5.2.14234" newVersion="1.5.2.14234" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Web.Helpers" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="1.0.0.0-3.0.0.0" newVersion="3.0.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Web.WebPages" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="0.0.0.0-3.0.0.0" newVersion="3.0.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Web.Mvc" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="0.0.0.0-5.2.0.0" newVersion="5.2.0.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
  <system.webServer>
    <validation validateIntegratedModeConfiguration="false" />
  </system.webServer>
  <entityFramework>
    <!--<contexts>
      <context type="ContosoUniversity.DAL.SchoolContext, ContosoUniversity">
        <databaseInitializer type="ContosoUniversity.DAL.SchoolInitializer, ContosoUniversity" />
      </context>
    </contexts>-->
    <defaultConnectionFactory type="System.Data.Entity.Infrastructure.LocalDbConnectionFactory, EntityFramework">
      <parameters>
        <parameter value="v11.0" />
      </parameters>
    </defaultConnectionFactory>
    <providers>
      <provider invariantName="System.Data.SqlClient" type="System.Data.Entity.SqlServer.SqlProviderServices, EntityFramework.SqlServer" />
    </providers>
  </entityFramework>
</configuration>
````
- [x] Update the `default.toml` to match:
```toml
port = 8099
app_pool = "contoso_pool"
app_name = "contoso_universtiy"
site_name = "contoso_site"
```
- [x] Update the run command instructions to include the port mapping:
```
docker run --env HAB_LICENSE="accept-no-persist" -it -p 80:8099 <origin>/<image>
```
- [x] Add instructions on how to access the website on your computer `http://localhost/contoso_university`                        

# Acceptance Criteria

1. A user is able to follow the instructions on a Windows computer and complete the tutorial without any issues.

### PR DESCRIPTION
**Pull request by dmccown**
The Contoso University demo is broken at this moment. This will ensure
someone can complete the demo without any issues.

Fixes #6783

### CODE REVIEW

File: www/source/demo/windows/steps/5.html.slim [hunk 1]
```diff
@@ -10,10 +10,89 @@ section
...
+    app_name = "contoso_university"
+    site_name = "contoso_site"
 
  p For more information on how to configure plans, see #{link_to 'Configuration Templates','/docs/developing-packages#add-configuration'}.
 
-= link_to 'Next: Leveraging Powershell DSC', "/demo/windows/steps/6", class: 'button cta'
\ No newline at end of file
+ h2 Create a web.config file
+
+ p Create a <code>web.config</code> file in the config fold. The contents of the file should be:
```
Comment (dmccown) [inline www/source/demo/windows/steps/5.html.slim]: Typo on fold.

File: www/source/demo/windows/steps/5.html.slim [hunk 2]
```diff
@@ -10,10 +10,89 @@ section
...
+        <defaultConnectionFactory type="System.Data.Entity.Infrastructure.LocalDbConnectionFactory, EntityFramework">
+          <parameters>
+            <parameter value="v11.0" />
+          </parameters>
+        </defaultConnectionFactory>
+        <providers>
+          <provider invariantName="System.Data.SqlClient" type="System.Data.Entity.SqlServer.SqlProviderServices, EntityFramework.SqlServer" />
+        </providers>
+      </entityFramework>
+    </configuration>
```
Comment (mwrock) [inline www/source/demo/windows/steps/5.html.slim]: Pasting in this `web.config` should not be necessary if following https://www.habitat.sh/demo/windows/steps/9/.

File: www/source/demo/windows/steps/7.html.slim [hunk 3]
```diff
@@ -14,6 +14,8 @@ section
      Set-Location {{pkg.svc_path}}
      if(Test-Path var) { Remove-Item var -Recurse -Force }
      New-Item -Name var -ItemType Junction -target "{{pkg.path}}/www" | Out-Null
+     Set-Location {{pkg.svc_path}}\var
+     New-Item -Name Web.config -ItemType SymbolicLink -target "{{pkg.svc_config_path}}/Web.config" -Force | Out-Null
```
Comment (mwrock) [inline www/source/demo/windows/steps/7.html.slim]: same as above.

Comment (dmccown) [reply www/source/demo/windows/steps/7.html.slim]: Yeah we run into these issues following steps on the demo that the `web.config` wasn't found when starting the application in the docker container. If you get this working as it is today let me know.

Comment (dmccown) [reply www/source/demo/windows/steps/7.html.slim]: Hmm you know what the confusion is. This step: https://www.habitat.sh/demo/windows/steps/8/#export-container suggests you should start your app before you do the work in step 9, so at this point your application will not start up. That's where some people running through the demo got stuck.

Comment (mwrock) [reply www/source/demo/windows/steps/7.html.slim]: Ah I see what happened here. At some point we changed the perms on the `config` directory and needed to add:
```
# The svc_config_path lacks an ACL for the USERS group
# so we need to ensure the app pool user can access those files
$pool = "{{cfg.app_pool}}"
$access = New-Object System.Security.AccessControl.FileSystemAccessRule `
  "IIS APPPOOL\$pool",`
  "ReadAndExecute",`
  "Allow"
$acl = Get-Acl "{{pkg.svc_config_path}}/Web.config"
$acl.SetAccessRule($access)
$acl | Set-Acl "{{pkg.svc_config_path}}/Web.config"
```
to the `run` hook. In the tutorial, we should have added that in step 9 and omitted it from step 7.

I'll can add that to this PR.

**Comment by dmccown** [comment]
I'm closing this PR in lieu of work @mwrock is doing to fix this issue.

**Pull request closed without merging**